File: sdk/php/src/Model/WsMessage.php
<?php

namespace KuCoin\UniversalSDK\Model;

use JMS\Serializer\Serializer;
use JMS\Serializer\Annotation\Exclude;
use JMS\Serializer\Annotation\SerializedName;
use JMS\Serializer\Annotation\Type;

/**
 * Class WsMessage
 * Represents a message between the WebSocket client and server.
 */
class WsMessage
{
    /**
     * Unique message ID
     * @var string
     * @Type("string")
     * @SerializedName("id")
     */
    public $id;

    /**
     * Message type (e.g., "ping", "subscribe", etc.)
     * @var string
     * @Type("string")
     * @SerializedName("type")
     */
    public $type;

    /**
     * Sequence number
     * @var int
     * @Type("integer")
     * @SerializedName("sn")
     */
    public $sn;

    /**
     * The topic of the message
     * @var string
     * @Type("string")
     * @SerializedName("topic")
     */
    public $topic;

    /**
     * Subject of the message
     * @var string
     * @Type("string")
     * @SerializedName("subject")
     */
    public $subject;

    /**
     * Indicates if it is a private channel
     * @var bool
     * @Type("boolean")
     * @SerializedName("privateChannel")
     */
    public $privateChannel;

    /**
     * Indicates if the message is a response
     * @var bool
     * @Type("boolean")
     * @SerializedName("response")
     */
    public $response;

    /**
     * Raw message data
     * @var mixed
     * @Type("JsonSerialized")
     * @SerializedName("data")
     */
    public $rawData;

    /**
     * @param string $json
     * @param Serializer $serializer
     * @return self
     */
    public static function jsonDeserialize(string $json, Serializer $serializer): WsMessage
    {
        return $serializer->deserialize($json, self::class, 'json');
    }

    /**
     * @param Serializer $serializer
     * @return string
     */
    public function jsonSerialize(Serializer $serializer): string
    {
        return $serializer->serialize($this, 'json');
    }
}

File: sdk/java/src/main/java/com/kucoin/universal/sdk/model/WsMessage.java
package com.kucoin.universal.sdk.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import lombok.ToString;

@Data
@ToString(exclude = "data")
@JsonIgnoreProperties(ignoreUnknown = true)
public class WsMessage<T> {

  /** Unique message ID */
  @JsonProperty("id")
  private String id;

  /** Message type (e.g., "ping", "subscribe", etc.) */
  @JsonProperty("type")
  private String type;

  /** Sequence number */
  @JsonProperty("sn")
  private Long sn;

  /** The topic of the message */
  @JsonProperty("topic")
  private String topic;

  /** Subject of the message */
  @JsonProperty("subject")
  private String subject;

  /** Indicates if it is a private channel */
  @JsonProperty("privateChannel")
  private Boolean privateChannel;

  /** Indicates if the message is a response */
  @JsonProperty("response")
  private Boolean response;

  /** Raw message data */
  @JsonProperty("data")
  private T data;
}
